docs/src/main/asciidoc/_migration-guide-for-4.0.adoc

```diff
@@ -904,6 +1018,38 @@ public class Demo {
 }
 ----
 
+==== From azure-spring-integration-servicebus to spring-integration-azure-servicebus
+This guide is intended to assist in the migration to
+link:https://search.maven.org/artifact/com.azure.spring/spring-integration-azure-servicebus[spring-integration-azure-servicebus] from
+version 2 of link:https://search.maven.org/artifact/com.azure.spring/azure-spring-integration-servicebus[azure-spring-integration-servicebus].
+
+* Please refer to <<migration-guide-introduction, introduction>> and <<migration-guide-benefits, migration benefits>> to get a whole picture of the changes in 4.0.
+* Please refer to <<migration-guide-naming, naming changes>> to learn more about the strategy changes in the project naming.
+* Please refer to <<migration-guide-bom, bom>> to learn how to using one BOM for all Spring Cloud Azure libraries.
+
+[#api-spring-integration-azure-servicebus]
+===== API Changes
+* Drop `ServiceBusQueueOperation` and `ServiceBusTopicOperation` with the subscribing function moved to class `ServiceBusMessageListenerContainer` and the sending function moved to `ServiceBusTemplate`.
+* Drop `ServiceBusQueueInboundChannelAdapter` and `ServiceBusTopicInboundChannelAdapter`, and combine the function of listening to a Service Bus queue or topic entity to `ServiceBusInboundChannelAdapter`.
+* Change the constructor from `ServiceBusXXXInboundChannelAdapter(String, SubscribeByGroupOperation, String)` to `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`.
+* Drop API `setCheckpointConfig` of ServiceBusTopicOperation or ServiceBusQueueOperation. To set the checkpoint configuration for the inbound channel adapter, users can call the method `ServiceBusContainerProperties#setAutoComplete` instead. To disable the auto-complete mode is equivalent to `MANUAL` checkpoint mode and to enable it will trigger the `RECORD` mode.
+* Drop API `setClientConfig` of `ServiceBusQueueOperation` and `ServiceBusTopicOperation`. To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
+* Drop `CompletableFuture` support of `ServiceBusTemplate` and `DefaultMessageHandler` and support Reactor instead.
+* Add new API of `setDefaultEntityType` for `ServiceBusTemplate` to specify the entity type, which is required when no bean of `PropertiesSupplier&lt;String, ProducerProperties&gt;` is provided for the `ProducerProperties#entityType`.
+* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
+
+.Class mapping from azure-spring-cloud-starter-servicebus to spring-cloud-azure-starter-integration-servicebus
```

Comment:
this table should be called `Class mapping from azure-spring-integration-servicebus to spring-integration-azure-servicebus`